standalone/pherry/src/chain_client.rs

```diff
@@ -70,59 +65,53 @@ pub async fn fetch_storage_changes(client: &XtClient, hash: &Hash) -> Result<Sto
 }
 
 /// Fetch the genesis storage.
-pub async fn fetch_genesis_storage(client: &XtClient) -> Result<Vec<(Vec<u8>, Vec<u8>)>> {
-    let hash = Some(*client.genesis());
-    let response = client.rpc.storage_pairs(StorageKey(vec![]), hash).await?;
+pub async fn fetch_genesis_storage(api: &ParachainApi) -> Result<Vec<(Vec<u8>, Vec<u8>)>> {
+    let hash = Some(*api.client.genesis());
+    let response = api
+        .client
+        .extra_rpc()
+        .storage_pairs(StorageKey(vec![]), hash)
+        .await?;
     let storage = response.into_iter().map(|(k, v)| (k.0, v.0)).collect();
     Ok(storage)
 }
 
 /// Fetch best next sequence for given sender considering the txpool
 pub async fn mq_next_sequence(
-    client: &XtClient,
+    api: &ParachainApi,
     sender: &MessageOrigin,
 ) -> Result<u64, subxt::Error> {
     let sender_scl = sender.encode();
     let sender_hex = hex::encode(sender_scl);
-    let seq: u64 = client
-        .rpc
+    let seq: u64 = api
+        .client
+        .rpc()
         .client
         .request("pha_getMqNextSequence", &[to_value(sender_hex)?])
         .await?;
     Ok(seq)
 }
 
-pub fn paras_heads_key(para_id: &ParaId) -> StorageKey {
-    StorageKey(storage_map_key_vec("Paras", "Heads", &para_id.encode()))
+pub fn paras_heads_key(para_id: u32) -> StorageKey {
+    let id = phaxt::kusama::runtime_types::polkadot_parachain::primitives::Id(para_id);
+    let entry = phaxt::kusama::paras::storage::Heads(id);
+    phaxt::storage_key(entry)
 }
 
-pub fn get_parachain_heads(head: Vec<u8>) -> Result<Vec<u8>, Error> {
+pub fn decode_parachain_heads(head: Vec<u8>) -> Result<Vec<u8>, Error> {
     Decode::decode(&mut head.as_slice()).or(Err(Error::FailedToDecode))
 }
 
 /// Updates the nonce from the mempool
-pub async fn update_signer_nonce(client: &XtClient, signer: &mut SrSigner) -> Result<()> {
+pub async fn update_signer_nonce(api: &ParachainApi, signer: &mut SrSigner) -> Result<()> {
     let account_id = signer.account_id().clone();
-    let nonce = client.account_nonce(&account_id).await?;
+    let nonce = api
+        .storage()
+        .system()
```

Comment:
`storage().system().account(...).await.nonce` is the on-chain nonce. The old `account_nonce()` api returns the nonce in the mempool (by the rpc `system_accountNextId`).

Not sure if it's a problem here but at least it doesn't match the comment of the function.

Reply:
Fixed